### PR TITLE
Add command to show report. Closes #8

### LINKED ISSUE #8: Add command to show report
It would be nice to have a command to show the report without having to reload everything.

### PR DESCRIPTION
**Pull request by jerone**
Add command `extensionUpdateReporter.openReport` to show Extension Update Report. 

Closes #8.

Instead of disposing the extensions store when closing the report, it now keeps the list of extensions in memory to allow re-opening the report per user command.

When there are no extension updates, the report looks like this:
image

### CODE REVIEW

**Comment by jerone** [comment]
@ryu1kn Let me know what you think about this implementation...

Plus, maybe you can help me with the two questions in the TODO.

**Comment by ryu1kn** [comment]
@jerone thank you for the draft and the ideas!

Implementation-wise, how you hooked the logic looks reasonable to me 👍 Not sure why we need to drop `context` though.

The 3rd point is interesting. I'd love to understand how it should behave first. Looking at the #8 description, 

* When we're running our vscode, completed extension update installation, (without restarting), we trigger this new command, what should we see?
  * Looking at the #8 description, perhaps we want to see the update report without restarting vscode. But at this stage, are the updated extensions already loaded and can be recognised by this extension?
* When we see the report with the new command (or at the vscode startup time), then close the report, then use the new command, what should we see?
  * The user might use the new command hoping that they can open the same report again to check something 🤔 
* As for the report screen, I don't mind too much if it ended up opening multiple reports. Every time we trigger the new command, it opens a new report, that is okay to me (although it may not look cool).

**Comment by jerone** [comment]
> Implementation-wise, how you hooked the logic looks reasonable to me 👍 Not sure why we need to drop `context` though.

We don't need the `context` anymore inside the `panel.onDidDispose()`, because we don't want to unsubscribe the command when webview is closed. 

> The 3rd point is interesting. I'd love to understand how it should behave first. Looking at the #8
> * When we're running our vscode, completed extension update installation, (without restarting), we trigger this new command, what should we see?
>   * Looking at the https://github.com/ryu1kn/vscode-extension-update-reporter/issues/8 description, perhaps we want to see the update report without restarting vscode. But at this stage, are the updated extensions already loaded and can be recognised by this extension?
> * When we see the report with the new command (or at the vscode startup time), then close the report, then use the new command, what should we see?
>   * The user might use the new command hoping that they can open the same report again to check something 🤔

As the request is to see the report, I would expect **to see the exact same report** I see when I start up VSCode.
The use-case is that the user might not have time at that moment to read all changelogs (he needs to do an quick bugfix first), and want to read them at a more convenient time (before closing VSCode).

**Comment by jerone** [comment]
> As the request is to see the report, I would expect **to see the exact same report** I see when I start up VSCode.

I just realized that this PR is not doing that, because of some debug code I had. Will investigate how to keep list of extension available, without creating a new setting.

**Comment by jerone** [comment]
@ryu1kn This is ready for review.

Changes:
* Report is now shown with all extension updates, until next reload.
* Added an custom message when people open the report when there were no extension updates.
* I added tests where possible.
* Added `VsCodeLike` type to resolve a few `any`s in the code.

**Comment by jerone** [comment]
Please squash this PR, before merging.

**Comment by jerone** [comment]
Possible future features:
-  Show date of last extensions check.
-  Add button to manually check again.

**Comment by jerone** [comment]
@ryu1kn Time to review the 3 open PR's... :shipit:

**Comment by jerone** [comment]
@ryu1kn 🏓

<small>Would love to be member of this repo...</small>

**Comment by jerone** [comment]
@ryu1kn Happy New Year 🥂🎆

Hopefully you have time to review these PR's in this new year...

 I really miss these useful features, but I don't want to create my own extension.

**Comment by jerone** [comment]
@ryu1kn, hope you are doing well.

4+ months passed since opening this PR (and others). Soon I'll be closing my PR's here and cloning my own extension. I don't want to do that, but I really want to these features, as I care about what extensions are doing on my machine.

Alternatively I would love to be member of this repo and move this along, if you are busy with other things.

**Comment by ryu1kn** [comment]
Hi @jerone , thank you very much for this enhancement. Very sorry that it took me long to get back to this PR, it must have been very frustrating / disappointing!

You've been improving this extension with multiple features + multiple things around tooling, and I’m not keeping up with your PR speed these days. I feel it makes sense that you approve PRs and merge them to the main branch yourself. I've just invited you as a collaborator 🙏 

I'd still like to do the release part for now so that I can sense check the code before it gets to the end users hands. It may not be perfect but being a collaborator should at least remove the pain of waiting for one PR approval while you have other improvements ideas on other PRs 👍

**Review by ryu1kn — DISMISSED** [review]

**Comment by ryu1kn** [comment]
@jerone Could you also update the CHANGELOG please?

**Comment by jerone** [comment]
Thanks @ryu1kn for adding me as collaborator! 
Sorry to have pinged you so much.
I'm fine with current setup. I really didn't want to publish my own extension to get these features.

I'll update and merge all PRs this weekend, so you can publish a new version after that.
How can I let you know?

**Comment by jerone** [comment]
> [..] I feel it makes sense that you approve PRs and merge them to the main branch yourself. [..]

Hmmm, due to the settings, I'm not allowed to approve my own PRs;
image
image

**Comment by ryu1kn** [comment]
>> [..] I feel it makes sense that you approve PRs and merge them to the main branch yourself. [..]
>
> Hmmm, due to the settings, I'm not allowed to approve my own PRs;

Apologies for the mis-configuration :| Now you should be able to merge PRs without approvals.

> I'll update and merge all PRs this weekend, so you can publish a new version after that.
> How can I let you know?

Good point. How about once you merge all the PRs you want to batch, you ping me on the last PR that now they're good to go? Then I'll update the CHANGELOG to give a date & version number (for this, if you could indicate if a release should be major / minor / patch version up, it would be great)

**Review by ryu1kn — APPROVED** [review]

**Comment by jerone** [comment]
_Will be misusing this PR for some communication :P_

> Apologies for the mis-configuration :| Now you should be able to merge PRs without approvals.

👍

> Good point. How about once you merge all the PRs you want to batch, you ping me on the last PR that now they're good to go? Then I'll update the CHANGELOG to give a date & version number (for this, if you could indicate if a release should be major / minor / patch version up, it would be great)

I'll create an release PR, which you then can update (new date if needed) and merge.

Also, would you setting enabling this option:
image

**Pull request merged**
